src/webserver/DOSGuard.h

```diff
@@ -103,15 +163,26 @@ class DOSGuard
     {
         if (whitelist_.contains(ip))
             return true;
-
         std::unique_lock lck(mtx_);
-        auto it = ipFetchCount_.find(ip);
-        if (it == ipFetchCount_.end())
-            return true;
+        bool fetchesOk =
+            (maxFetches_ == -1) ? true : getFetches(ip) < maxFetches_;
```

Comment:
I think `maxFetches_ == 1 || getFetches(ip) < maxFetches_` is cleaner, but up to you